delfin/drivers/dell_emc/vnx_block/navi_handler.py

```diff
@@ -0,0 +1,469 @@
+# Copyright 2020 The SODA Authors.
+# Copyright (c) 2016 Huawei Technologies Co., Ltd.
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+import copy
+import re
+import time
+
+import six
+from oslo_log import log as logging
+from oslo_utils import units
+
+from delfin import cryptor
+from delfin import exception
+from delfin.drivers.utils.navicli_client import NaviClient
+from delfin.drivers.utils.tools import Tools
+
+LOG = logging.getLogger(__name__)
+
+
+class NaviHandler(object):
+    """Common class for EMC VNX storage system."""
+
+    USER_SECURITY_API = 'naviseccli -AddUserSecurity -User %(username)s ' \
```

Comment:
naviseccli also supports file based authentication..will we support that too?

Reply:
yes

Reply:
You mean this code also support this

Reply:
naviseccli -AddUserSecurity -User
This sentence is used to generate file authentication

Reply:
ok